src/lib/Microsoft.Health.Events/EventCheckpointing/StorageCheckpointClient.cs

```diff
@@ -151,11 +154,34 @@ public async Task SetCheckpointAsync(IEventMessage eventArgs)
                 _checkpoints[partitionId] = checkpoint;
                 var count = _lastCheckpointTracker.AddOrUpdate(partitionId, 1, (key, value) => value + 1);
 
+                if (metrics != null)
+                {
+                    _postCheckpointMetrics.TryGetValue(partitionId, out var partitionExists);
+
+                    if (partitionExists == null)
+                    {
+                        _postCheckpointMetrics[partitionId] = new List<KeyValuePair<Metric, double>>();
+                    }
+
+                    foreach (var metric in metrics)
+                    {
+                        _postCheckpointMetrics[partitionId].Add(metric);
+                    }
+                }
+
                 if (count >= _lastCheckpointMaxCount)
                 {
                     await PublishCheckpointAsync(partitionId);
                     _log.LogMetric(EventMetrics.EventWatermark(partitionId, eventArgs.EnqueuedTime.UtcDateTime), 1);
                     _lastCheckpointTracker[partitionId] = 0;
+
+                    _postCheckpointMetrics.TryGetValue(partitionId, out var postCheckpointMetrics);
+                    postCheckpointMetrics?.ForEach(m =>
+                    {
+                        _log.LogMetric(m.Key, m.Value);
```

Comment:
I am fine with logging a metric for each batch and not trying to aggregate, just to keep things simple, since there would be different types of aggregations we would need to support (sum, average, stuff like that) and would rather something downstream handled that.